dataactbroker/handlers/fileHandler.py

```diff
@@ -411,48 +449,62 @@ def uploadFile(self):
 
-        submission_id = requestDict.getValue("submission_id")
-        start_date = requestDict.getValue("start")
-        end_date = requestDict.getValue("end")
+        Returns:
+            Tuple of boolean indicating successful start, and error response if False
 
-        if not (StringCleaner.isNumeric(submission_id) and StringCleaner.isDate(start_date) and StringCleaner.isDate(end_date)):
-            exc = ResponseException("submission id, start, and/or end cannot be parsed into their appropriate types", StatusCode.CLIENT_ERROR)
-            return JsonResponse.error(exc, exc.status)
+        """
+        jobDb = self.interfaces.jobDb
+        file_type_name = self.EXTERNAL_FILE_TYPE_MAP[file_type]
+
+        if file_type in ["D1", "D2"]:
+            # Populate start and end dates
+            requestDict = RequestDictionary(self.request)
+            start_date = requestDict.getValue("start")
+            end_date = requestDict.getValue("end")
+
+            if not (StringCleaner.isDate(start_date) and StringCleaner.isDate(end_date)):
+                exc = ResponseException("Start or end date cannot be parsed into a date", StatusCode.CLIENT_ERROR)
+                return False, JsonResponse.error(exc, exc.status, start = "", end = "", file_type = file_type, status = "failed")
+        elif file_type not in ["E","F"]:
+            exc = ResponseException("File type must be either D1, D2, E or F", StatusCode.CLIENT_ERROR)
+            return False, JsonResponse.error(exc, exc.status, file_type = file_type, status = "failed")
 
         cgac_code = self.jobManager.getSubmissionById(submission_id).cgac_code
-        get_url = CONFIG_BROKER["d1_url"].format(cgac_code, start_date, end_date)
 
         jq = JobQueue(job_queue_url=CONFIG_JOB_QUEUE['url'])
 
-        # Generate and upload D1 file to S3
+        # Generate and upload file to S3
         user_id = LoginSession.getName(session)
-        timestamped_name = s3UrlHandler.getTimestampedFilename(CONFIG_BROKER["award_procurement_file_name"])
+        timestamped_name = s3UrlHandler.getTimestampedFilename(CONFIG_BROKER["".join([str(file_type_name),"_file_name"])])
         upload_file_name = "".join([str(user_id), "/", timestamped_name])
-        d_file_id = self.jobManager.createDFileMeta(submission_id, start_date, end_date, "d1", CONFIG_BROKER["award_procurement_file_name"], upload_file_name)
-        job = jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, self.EXTERNAL_FILE_TYPE_MAP["D1"], "file_upload")
-        try:
-            job.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
-            job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
-        except ValueError as e:
-            # Date was not in expected format
-            raise ResponseException(str(e),StatusCode.CLIENT_ERROR,ValueError)
+
+        job = jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, self.EXTERNAL_FILE_TYPE_MAP[file_type_name], "file_upload")
```

Comment:
Related question about the dictionaries at the top of `fileHandler.py`...should the mapping of the file abbreviations (e.g., D2), be stored as part of the file_type table?